packages/blaze-query/src/hydra.ts

```diff
@@ -0,0 +1,758 @@
+    const utxos = await this.fetchUTxOs();
+
+    for (const key in utxos) {
+      const utxo = utxos[key]!;
+      if (utxo.output().amount().multiasset()?.has(unit)) {
+        return utxo;
+      }
+    }
+
+    throw new Error("UTxO not found");
+  }
+
+  async resolveUnspentOutputs(
+    txIns: TransactionInput[],
+  ): Promise<TransactionUnspentOutput[]> {
+    const utxos = await this.fetchUTxOs();
+
+    const ret: TransactionUnspentOutput[] = [];
+    for (const txIn of txIns) {
+      const utxo: TransactionUnspentOutput | undefined =
+        utxos[`${txIn.transactionId()}#${txIn.index()}`];
+      if (!utxo) {
+        throw new Error(
+          `Failed to resolve transaction: ${txIn.transactionId()}#${txIn.index()}`,
+        );
+      }
+
+      ret.push(utxo);
+    }
+
+    return ret;
+  }
+
+  /**
+   * This method takes a datum hash and tries to find the assosciated inline datum in the current snapshot utxo set.
+   * If the output containing the datum has been spent, this method will error
+   * @param datumHash
+   * @returns Promise<PlutusData>
+   */
+  async resolveDatum(datumHash: DatumHash): Promise<PlutusData> {
+    const utxos = await this.fetchUTxOs();
+
+    for (const key in utxos) {
+      const utxo = utxos[key]!;
+      const datum = utxo.output().datum();
+      if (!datum) continue;
+      if (datum.kind() != DatumKind.InlineData) continue;
+
+      const hash = datum.asInlineData()?.hash();
+      if (datumHash === hash) {
+        return datum.asInlineData()!;
+      }
+    }
+
+    throw new Error(
+      "resolveDatum: could not find a datum with that hash in the current snapshot",
+    );
+  }
+
+  async awaitTransactionConfirmation(
+    txId: TransactionId,
+    timeout?: number,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    const delay = (ms: number) =>
+      new Promise((resolve) => setTimeout(resolve, ms));
+
+    const checkConfirmation = async () => {
+      const resp = await fetch(`${this.url}snapshot/utxo`);
+      const rawUtxo = (await resp.json()) as any;
+      return !!Object.keys(rawUtxo).find(
+        (txRef) => txRef.split("#")[0] === txId,
+      );
+    };
+
+    if (await checkConfirmation()) {
+      return true;
+    }
+
+    if (timeout) {
+      while (Date.now() - startTime < timeout) {
+        await delay(500);
+
+        if (await checkConfirmation()) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  async postTransactionToChain(tx: BlazeTransaction): Promise<TransactionId> {
+    const txId = tx.body().hash();
+    this.connection.send(
+      JSON.stringify({
+        tag: "NewTx",
+        transaction: {
+          type: "Tx BabbageEra",
+          cborHex: tx.toCbor(),
+        },
+      }),
+    );
+
+    return txId;
+  }
+  evaluateTransaction(
+    _tx: BlazeTransaction,
+    _additionalUtxos: TransactionUnspentOutput[],
+  ): Promise<Redeemers> {
+    throw new Error("Unsupported by the Hydra API");
+  }
```

Comment:
This _could_ be provided. Anyone, please feel free to open a request here: https://github.com/cardano-scaling/hydra/issues

Reply:
Really we should eliminate this from all providers once web compatible vms are stable.